.github/workflows/check-upstream.yml

```diff
@@ -13,94 +13,89 @@ jobs:
 
   check-hash:
```

Comment:
Just a thought, maybe a "check hash" workflow can be the next todo item for the .github repository.